src/main/java/org/spongepowered/common/mixin/api/mcp/entity/EntityMixin_API.java

```diff
@@ -644,16 +641,26 @@ public EntityArchetype createArchetype() {
     }
 
     @Override
-    public org.spongepowered.api.data.value.Value.Mutable<Boolean> gravity() {
-        return this.getValue(Keys.HAS_GRAVITY).get();
+    public Set<Value.Immutable<?>> getValues() {
+        return this.api$getVanillaValues();
     }
 
-    protected void spongeApi$supplyVanillaManipulators(final Collection<? super Mutable<?, ?>> manipulators) {
-        this.get(VehicleData.class).ifPresent(manipulators::add);
-        if (this.fire > 0) {
-            manipulators.add(this.get(IgniteableData.class).get());
-        }
-        manipulators.add(new SpongeGravityData(!this.hasNoGravity()));
+    protected Set<Value.Immutable<?>> api$getVanillaValues() {
+        final Set<Value.Immutable<?>> values = new HashSet<>();
+
+        values.add(this.displayName().asImmutable());
+        values.add(this.passengers().asImmutable());
+        values.add(this.onGround().asImmutable());
+        values.add(this.velocity().asImmutable());
+        values.add(this.gravity().asImmutable());
+
+        this.baseVehicle().map(Value::asImmutable).ifPresent(values::add);
+        this.creator().map(Value::asImmutable).ifPresent(values::add);
+        this.notifier().map(Value::asImmutable).ifPresent(values::add);
+
+        this.getValue(Keys.FIRE_TICKS).map(Value::asImmutable).ifPresent(values::add);
+
```

Comment:
```suggestion
    if (this instanced CustomDataHolder) {
      values.add(((CustomDataHolder) this).data$gatherValues());
    }
```
I don't know if it's already a method, but if you want to try, that'd be the best way to do it imo, I'm sure @Zidane and @Cybermaxke will agree with me in that it'd be the easiest form of accessing the custom values/keys through that interface.

Reply:
Umm. Your going to need to explain that a bit more. 

You seem to mention custom values and keys but those are not within the scope of fixing the entities and adding their values into getValues

Reply:
@i509VCB Put the call he has above in getValues, you are correct.

Reply:
Put the call in getValues, you are correct